### PR TITLE
CON-4690 - Add php-cs-fixer

### PR DESCRIPTION
**Pull request by SimonBaeumer**
Implemented CS-Fixer:
- Install git hooks via `./.githooks/install-hooks.sh`
- Added `php-cs-fixer` to the build
- Removed `5.3` build, it only increases the build duration (wasn't the best idea at that moment)
- While commiting with coding style errors you get notifications which files break our coding style
- You can find more helpful commands here: https://github.com/FriendsOfPHP/PHP-CS-Fixer

### CODE REVIEW

File: Models/Connect/ConfigRepository.php [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+/**
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
 
 namespace   Shopware\CustomModels\Connect;
 
-use         \Shopware\Components\Model\ModelRepository;
+use         Shopware\Components\Model\ModelRepository;
```
Comment (sbofirov) [inline Models/Connect/ConfigRepository.php]: is it correct ?

File: .php_cs.dist [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?php
+$finder = PhpCsFixer\Finder::create()
+    ->exclude([ 'Library' ])
```
Comment (sbofirov) [inline .php_cs.dist]: do we need to exclude Views folder? It contains smarty or JS files.

File: .php_cs.dist [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+file that was distributed with this source code.
+EOF;
+
+return PhpCsFixer\Config::create()
+    ->setRules([
+        '@PSR2' => true,
+        'array_syntax' => ['syntax' => 'short'],
+        'blank_line_after_namespace' => true,
+        'blank_line_after_opening_tag' => true,
+        'blank_line_before_return' => true,
```
Comment (sbofirov) [inline .php_cs.dist]: it's deprecated, please use ```blank_line_before_statement```

File: .php_cs.dist (hunk 2 above)
Comment (SimonBaeumer) [inline .php_cs.dist]: No, the cs fixer will only fetch php files, so we only need to exclude source code which does not belong to us (or the project)

File: Models/Connect/ConfigRepository.php (hunk 1 above)
Comment (SimonBaeumer) [inline Models/Connect/ConfigRepository.php]: Yes

File: Components/BasketHelper.php [hunk 4]
```diff
@@ -645,7 +642,8 @@ public function findPercentagedIndividualVouchers($voucherCode)
             ->setParameter('voucherCode', $voucherCode);
 
 
-        return $builder->getQuery()->getResult();    }
+        return $builder->getQuery()->getResult();
```
Comment (SimonBaeumer) [inline Components/BasketHelper.php]: There is a statement which forces us to make a new line before every `return` statement but I cannot see any configuration for cases where a useless line was added... :/

File: .php_cs.dist (hunk 3 above)
Comment (SimonBaeumer) [inline .php_cs.dist]: I tried it and the php-cs-fixer crashes because he doesn't know that rule :/

File: .php_cs.dist (hunk 2 above)
Comment (sbofirov) [inline .php_cs.dist]: :+1:

File: .php_cs.dist (hunk 3 above)
Comment (sbofirov) [inline .php_cs.dist]: I saw that in their repository. Maybe we use different version.
It's not critical, so we can replace at some point in the future.

File: Components/BasketHelper.php (hunk 4 above)
Comment (sbofirov) [inline Components/BasketHelper.php]: me too :(

**Pull request merged**